### PR TITLE
Logger json

### PR DESCRIPTION
**Pull request by Integralist**
## Problem

We want to incorporate JSON formatted logging

## Solution

Implement the Alephant-Logger-Json gem

### CODE REVIEW

**Comment by darnould** [comment]
:dragon_face:

File: alephant-publisher-queue.gemspec [hunk 1]
```diff
@@ -34,7 +34,7 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'alephant-support'
   spec.add_runtime_dependency 'alephant-sequencer'
   spec.add_runtime_dependency 'alephant-cache'
-  spec.add_runtime_dependency 'alephant-logger', "1.2.0"
+  spec.add_runtime_dependency 'alephant-logger', "3.0.2"
```
Comment (revett) [inline alephant-publisher-queue.gemspec]: `>=`?

File: lib/alephant/publisher/env.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 require 'aws-sdk'
 require 'yaml'
+require 'alephant/logger'
```
Comment (revett) [inline lib/alephant/publisher/env.rb]: Quotes (Rubocop) - if you care

File: alephant-publisher-queue.gemspec (hunk 1 above)
Comment (darnould) [inline alephant-publisher-queue.gemspec]: We'd want `~>` to avoid breaking across major versions

File: lib/alephant/publisher/queue/options.rb [hunk 3]
```diff
@@ -52,7 +52,13 @@ def execute(instance, type, opts)
             instance.merge! opts
           rescue Exception => e
             logger.metric "QueueOptionsInvalidKeySpecified"
-            logger.error "Publisher::Queue::Options#validate: '#{e.message}'"
+            logger.error(
+              "event"     => "QueueOptionsKeyInvalid",
+              "class"     => e.class,
+              "message"   => e.message,
+              "backtrace" => e.backtrace.join("\n"),
```
Comment (revett) [inline lib/alephant/publisher/queue/options.rb]: `e.backtrace.to_s`

File: lib/alephant/publisher/queue/sqs_helper/archiver.rb [hunk 4]
```diff
@@ -39,8 +39,13 @@ def sync_store(message)
           end
 
           def store(message)
-            logger.info log_message(message)
-            store_item message
+            store_item(message).tap do
+              logger.info(
+                "event"  => "MessageStored",
+                "messageBody" => body_for(message),
+                "method" => "#{self.class}#store"
```
Comment (revett) [inline lib/alephant/publisher/queue/sqs_helper/archiver.rb]: Align `=>`

File: lib/alephant/publisher/queue/sqs_helper/archiver.rb [hunk 5]
```diff
@@ -39,8 +39,13 @@ def sync_store(message)
           end
 
           def store(message)
-            logger.info log_message(message)
-            store_item message
+            store_item(message).tap do
+              logger.info(
+                "event"  => "MessageStored",
+                "messageBody" => body_for(message),
```
Comment (revett) [inline lib/alephant/publisher/queue/sqs_helper/archiver.rb]: This uses `encoded_body()` which again will have the same problem as: 

```ruby
e.backtrace.join("\n")
```

File: lib/alephant/publisher/queue/sqs_helper/queue.rb [hunk 6]
```diff
@@ -38,7 +48,13 @@ def archive(m)
             archiver.see(m) unless archiver.nil?
           rescue StandardError => e
             logger.metric "ArchiveFailed"
-            logger.warn("Queue#archive: archive failed (#{e.message})");
+            logger.error(
+              "event"     => "MessageArchiveFailed",
+              "class"     => e.class,
+              "message"   => e.message,
+              "backtrace" => e.backtrace.join("\n"),
```
Comment (revett) [inline lib/alephant/publisher/queue/sqs_helper/queue.rb]: Same as above

File: lib/alephant/publisher/queue/writer.rb [hunk 7]
```diff
@@ -45,16 +45,42 @@ def perform
...
+              "method"    => "#{self.class}#write"
+            )
           end
         end
 
         def store(id, view, location, storage_opts = {})
-          logger.info "Publisher::Queue::Writer#store: location '#{location}', message.id '#{message.id}'"
-          cache.put(location, view.render, view.content_type, storage_opts)
-          lookup.write(id, options, seq_id, location)
+          render = view.render
```
Comment (revett) [inline lib/alephant/publisher/queue/writer.rb]: This doesn't seem necessary

**Comment by revett** [comment]
@Integralist added some comments, some are more important than others :+1:

File: lib/alephant/publisher/queue/sqs_helper/archiver.rb [hunk 8]
```diff
@@ -62,10 +67,8 @@ def log_message_parts(id)
             ]
           end
 
-          def log_message(m)
-            log_message_parts(m.id).tap do |parts|
-              parts << "(#{encoded_body m.body})" if log_message_body
-            end.join(" ")
+          def body_for(message)
+             log_message_body ? encoded_body(message.body) : "No message body available"
```
Comment (Integralist) [inline lib/alephant/publisher/queue/sqs_helper/archiver.rb]: @revett I'll just set this to `message.body` then?

File: lib/alephant/publisher/queue/writer.rb (hunk 7 above)
Comment (Integralist) [inline lib/alephant/publisher/queue/writer.rb]: @revett we use `render` further down as well though

**Pull request closed without merging**